Tests/DatadogTests/Datadog/TracerTests.swift

```diff
@@ -108,6 +108,33 @@ class TracerTests: XCTestCase {
         }
     }
 
+    func testSendingSpanWithGlobalTags() throws {
+        let server = ServerMock(delivery: .success(response: .mockResponseWith(statusCode: 200)))
+        TracingFeature.instance = .mockWorkingFeatureWith(
+            server: server,
+            directory: temporaryDirectory
+        )
+        defer { TracingFeature.instance = nil }
+
+        let tracer = Tracer.initialize(
+            configuration: .init(
+                serviceName: "custom-service-name",
+                globalTags: [
+                    "globaltag1": "globalValue1",
+                    "globaltag2": "globalValue2"
+                ]
+            )
+        )
+
+        let span = tracer.startSpan(operationName: .mockAny())
```

Comment:
We could also test that tags are properly overwritten with span-local tag:
```swift
span.setTag(key: "globaltag2", value: "overwritten value")

// ...

XCTAssertEqual(try spanMatcher.meta.custom(keyPath: "meta.globaltag2"), "overwritten value")
```
WDYT?

Reply:
Good idea, it covers the logic better. Changed